src/components/DayOverview/DayOverview.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import supabase from "@/supabaseClient";
+import DayOverviewButton from "@/components/DayOverview/DayOverviewButton";
+import { Schema } from "@/database_utils";
+
+interface Props {
+    text: string;
+    date: Date;
+    location: string;
+}
+
+export type ParcelOfSpecificDateAndLocation = Pick<Schema["parcels"], "collection_datetime"> & {
+    clients: Pick<
+        Schema["clients"],
+        "flagged_for_attention" | "full_name" | "address_postcode" | "delivery_instructions"
+    > | null;
+};
+
+export type DayOverviewData = {
+    date: Date;
+    location: string;
+    data: ParcelOfSpecificDateAndLocation[];
+};
+
+export const getCurrentDate = (date: Date, hyphen: boolean = false): string => {
+    const year = date.getFullYear();
+    const month = (date.getMonth() + 1).toString().padStart(2, "0");
+    const day = date.getDate().toString().padStart(2, "0");
+
+    return hyphen ? `${year}-${month}-${day}` : `${year}${month}${day}`;
+};
+
+const getParcelsOfSpecificDateAndLocation = async (
+    date: Date,
+    collectionCentre: string
+): Promise<ParcelOfSpecificDateAndLocation[]> => {
+    const startDateString = date.toISOString();
+    const endDate = new Date(date);
+    endDate.setDate(date.getDate() + 1);
+    const endDateString = endDate.toISOString();
+
+    const { data, error } = await supabase
+        .from("parcels")
+        .select(
+            "collection_datetime, clients(flagged_for_attention, full_name, address_postcode, delivery_instructions)"
+        )
+        .gte("collection_datetime", startDateString)
+        .lt("collection_datetime", endDateString)
+        .eq("collection_centre", collectionCentre)
+        .order("collection_datetime");
+
+    if (error) {
```

Comment:
might be worth either adding a TODO for VFB 22 or changing this to fit the new error handling messages - depending on which gets merged first ^^

Reply:
Done